cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -43,6 +51,24 @@ runTxSign (TxBodyFile infile) skfiles  network (TxFile outfile) = do
       . writeTxSigned outfile
       $ signTransaction txu network sks
 
+runTxSubmit :: FilePath -> ConfigYamlFilePath -> SocketPath -> ExceptT CliError IO ()
+runTxSubmit txFp configFp sktFp =
+  withIOManagerE $ \iocp -> do
+    nc <- liftIO $ parseNodeConfigurationFP configFp
+    SomeConsensusProtocol p <- firstExceptT ProtocolError $ mkConsensusProtocol nc Nothing
+    signedTx <- firstExceptT CardanoApiError . newExceptT $ readTxSigned txFp
+    case p of
+      Consensus.ProtocolRealTPraos{} -> do
+        let config = pInfoConfig $ Consensus.protocolInfo p
+        liftIO $ submitTx
+                   nullTracer -- tracer needed
+                   iocp
+                   config
+                   sktFp
+                   (prepareTxShelley signedTx)
+      _ -> panic "Wrong protocol specified"
```

Comment:
We _could_ use the `CliError` constructor, `IncorrectProtocolSpecifiedError`, here.